Basic-Car-Maintenance/Shared/Odometer/Views/EditOdometerReadingView.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  EditOdometerReadingView.swift
+//  Basic-Car-Maintenance
+//
+//  Created by Shakhnoza Mirabzalova on 10/26/23.
+//
+
+import FirebaseAnalyticsSwift
+import SwiftUI
+
+struct EditOdometerReadingView: View {
+    
+    @Binding var selectedOdometerReading: OdometerReading?
+    var viewModel: OdometerViewModel
+    @State private var date = Date()
+    @State private var selectedVehicle: Vehicle?
+    @State private var isMetric = false
+    @State private var distance = 0
+    @Environment(\.dismiss) var dismiss
+    
+    var body: some View {
+        NavigationStack {
+            Form {
+                Section {
+                    HStack {
+                        TextField("Distance", value: $distance, format: .number)
+                        
+                        Picker(selection: $isMetric) {
+                            Text("Miles").tag(false)
+                            Text("Kilometers").tag(true)
+                        } label: {
+                            Text("Preferred units", comment: "Label for units selected when adding an odometer reading")
+                        }
+                        .pickerStyle(.segmented)
+                    }
+                }
+                        Section {
+                            Picker(selection: $selectedVehicle) {
+                                ForEach(viewModel.vehicles) { vehicle in
+                                    Text(vehicle.name).tag(vehicle as Vehicle)
```

Comment:
The indentation of this section is off. Fix it with `Ctrl` + `i`

And this picker does not work. See the `AddMaintenanceView` for the proper way to set up the picker

Reply:
I revised the picker as per your suggestion, and now I can select the vehicle. However, I noticed that the odometer number remains the same. My understanding is that the purpose of having the picker is to display all the values based on the selected vehicle, right?

Wouldn't it be more logical to display the name of the vehicle instead of using the picker, especially since we're editing a specific item in the odometer view? 
I believe there should be a way to update all the values based on the selected vehicle, but I couldn't figure it out.

Attached are the screen recordings for the view with the picker and with just a name of the vehicle.

https://github.com/mikaelacaron/Basic-Car-Maintenance/assets/106402127/a4d48c82-31bd-428e-a47c-89db8c7f9fa7

https://github.com/mikaelacaron/Basic-Car-Maintenance/assets/106402127/f5ab9203-ed7c-403e-b585-c44b1fd7216b

Reply:
> My understanding is that the purpose of having the picker is to display all the values based on the selected vehicle, right?

No. The purpose of the picker is for some reason you wanted to change which vehicle this reading is for

> Wouldn't it be more logical to display the name of the vehicle instead of using the picker, especially since we're editing a specific item in the odometer view?

No, the goal is to change any value for this reading.

> I believe there should be a way to update all the values based on the selected vehicle, but I couldn't figure it out.

No that is not correct. Do not do this.

---

The purpose of this view is to be able to change any of the values for this specific reading. The first video you showed does not work properly because the vehicle does not change when you changed it in the picker. The 2nd video is not the correct behavior, because I **do** want the user to be able to change the vehicle.

Reply:
I've set the vehicle.id as the tag for the Picker(selection: $selectedVehicleID) to ensure that the selection works for the vehicles in the picker.
```
Picker(selection: $selectedVehicleID) {
            ForEach(vehicles) { vehicle in
              Text(vehicle.name).tag(vehicle.id)
            }
```
However, I encountered a challenge in updating the odometer reading based on the selected vehicle in the picker. The editOdometerReadingView is effectively updating the odometer for the vehicle listed in the OdometerView.

It seems that the issue might be related to the use of `@Binding var selectedOdometerReading: OdometerReading`. This variable is utilized in both the onAppear and the Button responsible for updating the odometer reading.

I've tried alternative approaches without the use of `@Binding var selectedOdometerReading: OdometerReading`, but I encountered some errors. I'd like to find the solution for the issue as well. As a newcomer to iOS development, I might've overlooked certain aspects, and I appreciate your understanding and patience.

Reply:
> I've set the `vehicle.id` as the tag for the `Picker(selection: $selectedVehicleID)` to ensure that the selection works for the vehicles in the picker

Can you commit this and push the changes? So I can see what you're doing on GitHub which can help me point you in the right direction

> but I encountered some errors

Can you elaborate more what the error were and what line

Look at the `EditMaintenanceEventView` and the `DashboardView` for how this works, you're doing essentially the same thing, but with odometer readings

Reply:
Hi @mikaelacaron
I pushed the commit from my local branch to dev. Can you please check if you can see it?